pkg/config/config_template.yaml

```diff
@@ -1,857 +1,1238 @@
+## and mount it in the container if needed.
+## If left empty, the CRI check is disabled.
+## see: https://docs.datadoghq.com/integrations/cri/
 #
-# To activate the CRI check you'll need to indicate the path of the
-# CRI runtime you're using (and mount it in the container if needed)
-# cri_socket_path: /var/run/containerd/containerd.sock
+# cri_socket_path: ""
+
+## @param cri_connection_timeout - integer - optional - default: 5
+## Configure the initial connection timeout in seconds.
 #
-# You can configure the initial connection timeout (in seconds)
-# cri_connection_timeout: 1
+# cri_connection_timeout: 5
+
+## @param cri_query_timeout - integer - optional - default: 5
+## Configure the timeout in seconds for querying the CRI.
 #
-# You can configure the timeout (in seconds) for querying the CRI
 # cri_query_timeout: 5
-#
+
 {{ end -}}
 {{- if .Containerd}}
-# Containerd integration
+
+##########################################
+## Containerd integration Configuration ##
+##########################################
+
+## @param cri_socket_path - string - optional - default: /var/run/containerd/containerd.sock
+## To activate the Containerd check, indicate the path of the Containerd socket you're using
+## and mount it in the container if needed.
+## see: https://docs.datadoghq.com/integrations/containerd/
 #
-# To activate the Containerd check you'll need to indicate the path of the
-# Containerd socket you're using (and mount it in the container if needed)
 # cri_socket_path: /var/run/containerd/containerd.sock
+
+## @param cri_query_timeout - integer - optional - default: 5
+## Configure the timeout in seconds for querying the Containerd API.
 #
-# You can configure the timeout (in seconds) for querying the Containerd API
 # cri_query_timeout: 5
+
+## @param containerd_namespace - string - optional - default: k8s.io
+## Activating the Containerd check also activates the CRI check, as it contains an additional subset of useful metrics.
+## Specify here the namespace that Containerd is using on your system. As the Containerd check
+## only supports Kubernetes, the default value is `k8s.io`
+## https://github.com/containerd/cri/blob/release/1.2/pkg/constants/constants.go#L22-L23
 #
-# Activating the Containerd check will also activate the CRI check, as it contains an additional subset of useful metrics.
-#
-# You can specify the namespace that Containerd is using on your system.
-# As the Containerd check only supports Kubernetes, the default value is `k8s.io`
-# https://github.com/containerd/cri/blob/release/1.2/pkg/constants/constants.go#L22-L23
 # containerd_namespace: k8s.io
+
 {{ end -}}
 {{- if .Kubelet }}
-# Kubernetes kubelet connectivity
+
+###################################################
+## Kubernetes kubelet connectivity Configuration ##
+###################################################
+
+## @param kubernetes_kubelet_host - string - optional
+## The kubelet host should be autodet
```

Comment:
`max_proc_fds` is missing in the new file.

Reply:
On the other hand, I can't see it being used in the code, nor `max_per_message`.

Are these not used, or am I missing something?

Reply:
this parameter is indeed not in use anymore